crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -561,6 +570,59 @@ impl<'a> TimelineEventHandler<'a> {
         }
     }
 
+    // Redacted redactions are no-ops (unfortunately)
+    #[instrument(skip_all, fields(redacts_event_id = ?redacts))]
+    fn handle_local_redaction(
+        &mut self,
+        redacts: OwnedTransactionId,
+        _content: RoomRedactionEventContent,
+    ) {
+        if let Some((_, rel)) = self.reaction_map.remove(&(Some(redacts.clone()), None)) {
+            update_timeline_item!(self, &rel.event_id, "redaction", |event_item| {
+                let Some(remote_event_item) = event_item.as_remote() else {
+                    error!("inconsistent state: redaction received on a non-remote event item");
+                    return None;
+                };
+
+                let mut reactions = remote_event_item.reactions.clone();
+
+                let count = {
+                    let Entry::Occupied(mut group_entry) = reactions.entry(rel.key.clone()) else {
+                        return None;
+                    };
+                    let group = group_entry.get_mut();
+
+                    if group.0.remove(&(Some(redacts.clone()), None)).is_none() {
+                        error!(
+                            "inconsistent state: reaction from reaction_map not in reaction list \
+                             of timeline item"
+                        );
+                        return None;
+                    }
+
+                    group.len()
+                };
+
+                if count == 0 {
+                    reactions.remove(&rel.key);
+                }
+
+                trace!("Removing reaction");
+                Some(event_item.with_kind(remote_event_item.with_reactions(reactions)))
+            });
+        }
+
+        // TODO: is the following still relevant?
+        // Even if the event being redacted is a reaction (found in
+        // `reaction_map`), it can still be present in the timeline items
+        // directly with the raw event timeline feature (not yet implemented).
+
+        if self.result.items_updated == 0 {
+            // We will want to know this when debugging redaction issues.
+            debug!("redaction affected no event");
```

Comment:
I think this should even be an `error!`, since local redactions that affect nothing would be an error on the part of the user of the timeline API.

Reply:
[`d92f11f`](https://github.com/matrix-org/matrix-rust-sdk/pull/2118/commits/d92f11f21c56d5e07e9c33a0dadd149f1cd72fbc)